interfaces/code/Profiler.java

```diff
@@ -0,0 +1,110 @@
+import java.util.Vector;
+
+/**
+ * This is a list implementation that likes to chat with you!
+ */
+class TalkativeList<E> extends ArrayList<E> {
+
+    // Since our super class (ArrayList) already implemented this method, we must override it!
+    @Override
+    public boolean contains(Object o) {
+
+        // We just print some information and then pass this request onto our super class!
+        System.out.printf("Looking for %s...\n", o.toString());
+
+        boolean contains = super.contains(o);
+
+        if (contains) {
+            System.out.println("I found it!");
+        } else {
+            System.out.println("No where to be seen.");
+        }
+
+        return contains;
+
+    }
+
+}
+
+class Profiler {
+
+    /**
+     * Prints out the time it takes for a list to perform a given set of operations.
+     */
+    public static void time_list(List<Integer> list, String name, int length) {
+
+        System.out.printf("Timing %s:\n", name);
+
+        // First, we fill the list with integers.
+        long start_time = System.currentTimeMillis();
+
+        for (int i = 0; i < length; i++) {
+            list.add(i);
+        }
+
+        long end_time = System.currentTimeMillis();
+        System.out.printf("\tFilling List: %dms\n", end_time - start_time);
+
+        // Now, we will iterate the list and increment every integer.
+        start_time = System.currentTimeMillis();
+
+        for (int i = 0; i < length; i++) {
+
+            // Retrieve the value.
+            int value = list.get(i);
+
+            // Now set it back into the list.
+            list.set(i, value + 1);
+
+        }
+
+        end_time = System.currentTimeMillis();
+        System.out.printf("\tIncrementing List: %dms\n", end_time - start_time);
+
+        // lastly, we will check that this list contains certain elements.
+        start_time = System.currentTimeMillis();
+
+        boolean contains_1 = list.contains(1);
+        boolean contains_5 = list.contains(5);
+        boolean contains_length = list.contains(length);
+        boolean contains_length_add_1 = list.contains(length + 1);
+
+        end_time = System.currentTimeMillis();
+        System.out.printf("\tSearching List: %dms\n", end_time - start_time);
+        System.out.printf(
+            "\t\tFound: 1=%b, 5=%b, %d=%b, %d=%b\n",
+            contains_1,
+            contains_5,
+            length,
+            contains_length,
+            length + 1,
+            contains_length_add_1
+        );
+
+    }
+
+    /**
+     *
+     * This will run a simple profiler that times different implementations of the list interface.
+     *
+     */
+    public static void main(String[] args) {
+
+        int list_length = 100000;
+
+        time_list(new ArrayList<>(), "ArrayList", list_length);
+        time_list(new LinkedList<>(), "LinkedList", list_length);
+        time_list(new Vector<>(), "Vector", list_length);
+        time_list(new TalkativeList<>(), "TalkativeList", list_length);
```

Comment:
Loving how easy it is to profile a new implementation. I hope the class appreciates it!